addon/mixins/route.js

```diff
@@ -287,6 +289,9 @@ export default Ember.Mixin.create({
    */
   updateInfinityModel(newObjects) {
     var infinityModel = this.get(this.get('_modelPath'));
+    if (infinityModel === undefined || newObjects === undefined) {
```

Comment:
It's sufficient to check for infinityModel being undefined here (or you need to add another test-case which explains the checking of newObjects).